### PR TITLE
added support for c++17 compilers

### PR DESCRIPTION
**Pull request by acki-m**
a lot of warning poped up from g++7
"will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]"
These can be remove when compiling with c++17 compiler flag

Additionally: VS2015 and greater will now use noexcept, was not enabled by accident
The benchmarks has to be disabled, because of compile errors in noniuose (no warnings)
https://github.com/libnonius/nonius/issues/97

### CODE REVIEW

**Comment by gabyx** [comment]
can we merge my disable-warning PR #100 improvements together with this one, and work on this in one PR? otherwise my work is kind of lost...

**Comment by gabyx** [comment]
Strangly clang++ 7.0 does not define
```
#if __has_feature(__cxx_noexcept_function_type)
#error("asd")
#endif
int main()
{
}
``` 
This does not compile as it should. also clang documentation does not say anythin about it: https://clang.llvm.org/docs/LanguageExtensions.html#langext-has-feature-back-compat

**Comment by gabyx** [comment]
I made some progress building, merged your PR #107 
clang 7.0 builds now everthing including benchmarks, and they work.
However as clang7.0 does not define `__cxx_noexcept_function_type`I had to make a hack in `core_prerequisites.h`.

**Comment by acki-m** [comment]
@gabyx 
because it this macro does not exist in clang
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4518.html
> The recommended feature-test macro is __cpp_noexcept_function_type. 

We have to use the one without `xx`

**Comment by gabyx** [comment]
hm ah :) I only looked on cpp reference :). crazy how much one needs to know to make it feasible. have you already thought about de flags? in the video he says cmake should find out what compiler flags one should use? I have no clue how he meant it. I mean to compile the library one needs at least -std=...., he also mentioned not to use Werror...
Is there a way to give a set of compile flags such that when cmake finds the library and import its target they are reasonable and make sense. So far nothing is imposed when you use find_package? i think this is good so isnt it?

**Comment by gabyx** [comment]
looks very good!

**Comment by gabyx** [comment]
I think my branch gets obsolete, which is good =)

**Comment by acki-m** [comment]
@gabyx 
The MacOSX builds are back. ^^
Do you know why the code on these machines don't compile?

```
/Users/travis/build/rttrorg/rttr/src/rttr/detail/misc/standard_types.cpp:96:18: error: no matching member function for call to 'method'
                .method("data",         &std::string::data)
                ~^~~~~~
/Users/travis/build/rttrorg/rttr/src/rttr/../rttr/registration.h:252:58: note: candidate template ignored: couldn't infer template argument 'F'
            bind<detail::meth, Class_Type, F, acc_level> method(string_view name, F f, acc_level level = acc_level());
```
https://travis-ci.org/rttrorg/rttr/jobs/335371216
https://travis-ci.org/rttrorg/rttr/jobs/335371221

**Comment by gabyx** [comment]
I will have a look tonight, I was experiencing yesterday some problems, I think I had an old installation, I compile with your branch again, install it and try to use it. With clang7.0 I get these kind of problems you are seeing will investigate! Thanks for the massive improvements :-)

**Comment by gabyx** [comment]
I merged all your change, I still left my set_warning_levels to be able to compile everything also the benchmarks and tests.
with clang7.0 it works fine on my macOs machine. I will test Apple Clang, supposedly it will break

**Comment by acki-m** [comment]
Now, it should build all linux targets fine (when travis will retrieve the package)
I think the apple clang issues are coming from different clang version scheme or mixing older standard libraries and new compiler version.

**Comment by gabyx** [comment]
we have some problems with (__clang__ ist not enough I think on macos we need to detect if its the apple clang and then check for the version (https://travis-ci.org/rttrorg/rttr/jobs/335840657) (sucks...) 
```
/Users/travis/build/rttrorg/rttr/src/rttr/../rttr/detail/misc/utility.h:44:1: error: unknown warning group '-Wnoexcept-type', ignored [-Werror,-Wunknown-pragmas]
RTTR_BEGIN_DISABLE_EXCEPT_TYPE_WARNING
^
/Users/travis/build/rttrorg/rttr/src/rttr/../rttr/detail/base/core_prerequisites.h:284:57: note: expanded from macro 'RTTR_BEGIN_DISABLE_EXCEPT_TYPE_WARNING'
                                                        _Pragma ("clang diagnostic ignored \"-Wnoexcept-type\"")
                                                        ^
<scratch space>:137:27: note: expanded from here
 clang diagnostic ignored "-Wnoexcept-type"
```

**Comment by gabyx** [comment]
may be we need something along the line:
`#ifdef __apple_build_version__ > ?????`

```
gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
gcc -x c /dev/null -dM -E
#define OBJC_NEW_PROPERTIES 1
#define _LP64 1
#define __APPLE_CC__ 6000
#define __APPLE__ 1
#define __ATOMIC_ACQUIRE 2
#define __ATOMIC_ACQ_REL 4
#define __ATOMIC_CONSUME 1
#define __ATOMIC_RELAXED 0
#define __ATOMIC_RELEASE 3
#define __ATOMIC_SEQ_CST 5
#define __BIGGEST_ALIGNMENT__ 16
#define __BLOCKS__ 1
#define __BYTE_ORDER__ __ORDER_LITTLE_ENDIAN__
#define __CHAR16_TYPE__ unsigned short
#define __CHAR32_TYPE__ unsigned int
#define __CHAR_BIT__ 8
#define __CONSTANT_CFSTRINGS__ 1
#define __DBL_DECIMAL_DIG__ 17
#define __DBL_DENORM_MIN__ 4.9406564584124654e-324
#define __DBL_DIG__ 15
#define __DBL_EPSILON__ 2.2204460492503131e-16
#define __DBL_HAS_DENORM__ 1
#define __DBL_HAS_INFINITY__ 1
#define __DBL_HAS_QUIET_NAN__ 1
#define __DBL_MANT_DIG__ 53
#define __DBL_MAX_10_EXP__ 308
#define __DBL_MAX_EXP__ 1024
#define __DBL_MAX__ 1.7976931348623157e+308
#define __DBL_MIN_10_EXP__ (-307)
#define __DBL_MIN_EXP__ (-1021)
#define __DBL_MIN__ 2.2250738585072014e-308
#define __DECIMAL_DIG__ __LDBL_DECIMAL_DIG__
#define __DYNAMIC__ 1
#define __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ 101100
#define __FINITE_MATH_ONLY__ 0
#define __FLT_DECIMAL_DIG__ 9
#define __FLT_DENORM_MIN__ 1.40129846e-45F
#define __FLT_DIG__ 6
#define __FLT_EPSILON__ 1.19209290e-7F
#define __FLT_EVAL_METHOD__ 0
#define __FLT_HAS_DENORM__ 1
#define __FLT_HAS_INFINITY__ 1
#define __FLT_HAS_QUIET_NAN__ 1
#define __FLT_MANT_DIG__ 24
#define __FLT_MAX_10_EXP__ 38
#define __FLT_MAX_EXP__ 128
#define __FLT_MAX__ 3.40282347e+38F
#define __FLT_MIN_10_EXP__ (-37)
#define __FLT_MIN_EXP__ (-125)
#define __FLT_MIN__ 1.17549435e-38F
#define __FLT_RADIX__ 2
#define __FXSR__ 1
#define __GCC_ATOMIC_BOOL_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __GCC_ATOMIC_CHAR_LOCK_FREE 2
#define __GCC_ATOMIC_INT_LOCK_FREE 2
#define __GCC_ATOMIC_LLONG_LOCK_FREE 2
#define __GCC_ATOMIC_LONG_LOCK_FREE 2
#define __GCC_ATOMIC_POINTER_LOCK_FREE 2
#define __GCC_ATOMIC_SHORT_LOCK_FREE 2
#define __GCC_ATOMIC_TEST_AND_SET_TRUEVAL 1
#define __GCC_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_16 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8 1
#define __GNUC_MINOR__ 2
#define __GNUC_PATCHLEVEL__ 1
#define __GNUC_STDC_INLINE__ 1
#define __GNUC__ 4
#define __GXX_ABI_VERSION 1002
#define __GXX_RTTI 1
#define __INT16_C_SUFFIX__
#define __INT16_FMTd__ "hd"
#define __INT16_FMTi__ "hi"
#define __INT16_MAX__ 32767
#define __INT16_TYPE__ short
#define __INT32_C_SUFFIX__
#define __INT32_FMTd__ "d"
#define __INT32_FMTi__ "i"
#define __INT32_MAX__ 2147483647
#define __INT32_TYPE__ int
#define __INT64_C_SUFFIX__ LL
#define __INT64_FMTd__ "lld"
#define __INT64_FMTi__ "lli"
#define __INT64_MAX__ 9223372036854775807LL
#define __INT64_TYPE__ long long int
#define __INT8_C_SUFFIX__
#define __INT8_FMTd__ "hhd"
#define __INT8_FMTi__ "hhi"
#define __INT8_MAX__ 127
#define __INT8_TYPE__ signed char
#define __INTMAX_C_SUFFIX__ L
#define __INTMAX_FMTd__ "ld"
#define __INTMAX_FMTi__ "li"
#define __INTMAX_MAX__ 9223372036854775807L
#define __INTMAX_TYPE__ long int
#define __INTMAX_WIDTH__ 64
#define __INTPTR_FMTd__ "ld"
#define __INTPTR_FMTi__ "li"
#define __INTPTR_MAX__ 9223372036854775807L
#define __INTPTR_TYPE__ long int
#define __INTPTR_WIDTH__ 64
#define __INT_FAST16_FMTd__ "hd"
#define __INT_FAST16_FMTi__ "hi"
#define __INT_FAST16_MAX__ 32767
#define __INT_FAST16_TYPE__ short
#define __INT_FAST32_FMTd__ "d"
#define __INT_FAST32_FMTi__ "i"
#define __INT_FAST32_MAX__ 2147483647
#define __INT_FAST32_TYPE__ int
#define __INT_FAST64_FMTd__ "ld"
#define __INT_FAST64_FMTi__ "li"
#define __INT_FAST64_MAX__ 9223372036854775807L
#define __INT_FAST64_TYPE__ long int
#define __INT_FAST8_FMTd__ "hhd"
#define __INT_FAST8_FMTi__ "hhi"
#define __INT_FAST8_MAX__ 127
#define __INT_FAST8_TYPE__ signed char
#define __INT_LEAST16_FMTd__ "hd"
#define __INT_LEAST16_FMTi__ "hi"
#define __INT_LEAST16_MAX__ 32767
#define __INT_LEAST16_TYPE__ short
#define __INT_LEAST32_FMTd__ "d"
#define __INT_LEAST32_FMTi__ "i"
#define __INT_LEAST32_MAX__ 2147483647
#define __INT_LEAST32_TYPE__ int
#define __INT_LEAST64_FMTd__ "ld"
#define __INT_LEAST64_FMTi__ "li"
#define __INT_LEAST64_MAX__ 9223372036854775807L
#define __INT_LEAST64_TYPE__ long int
#define __INT_LEAST8_FMTd__ "hhd"
#define __INT_LEAST8_FMTi__ "hhi"
#define __INT_LEAST8_MAX__ 127
#define __INT_LEAST8_TYPE__ signed char
#define __INT_MAX__ 2147483647
#define __LDBL_DECIMAL_DIG__ 21
#define __LDBL_DENORM_MIN__ 3.64519953188247460253e-4951L
#define __LDBL_DIG__ 18
#define __LDBL_EPSILON__ 1.08420217248550443401e-19L
#define __LDBL_HAS_DENORM__ 1
#define __LDBL_HAS_INFINITY__ 1
#define __LDBL_HAS_QUIET_NAN__ 1
#define __LDBL_MANT_DIG__ 64
#define __LDBL_MAX_10_EXP__ 4932
#define __LDBL_MAX_EXP__ 16384
#define __LDBL_MAX__ 1.18973149535723176502e+4932L
#define __LDBL_MIN_10_EXP__ (-4931)
#define __LDBL_MIN_EXP__ (-16381)
#define __LDBL_MIN__ 3.36210314311209350626e-4932L
#define __LITTLE_ENDIAN__ 1
#define __LONG_LONG_MAX__ 9223372036854775807LL
#define __LONG_MAX__ 9223372036854775807L
#define __LP64__ 1
#define __MACH__ 1
#define __MMX__ 1
#define __NO_INLINE__ 1
#define __NO_MATH_INLINES 1
#define __ORDER_BIG_ENDIAN__ 4321
#define __ORDER_LITTLE_ENDIAN__ 1234
#define __ORDER_PDP_ENDIAN__ 3412
#define __PIC__ 2
#define __POINTER_WIDTH__ 64
#define __PRAGMA_REDEFINE_EXTNAME 1
#define __PTRDIFF_FMTd__ "ld"
#define __PTRDIFF_FMTi__ "li"
#define __PTRDIFF_MAX__ 9223372036854775807L
#define __PTRDIFF_TYPE__ long int
#define __PTRDIFF_WIDTH__ 64
#define __REGISTER_PREFIX__
#define __SCHAR_MAX__ 127
#define __SHRT_MAX__ 32767
#define __SIG_ATOMIC_MAX__ 2147483647
#define __SIG_ATOMIC_WIDTH__ 32
#define __SIZEOF_DOUBLE__ 8
#define __SIZEOF_FLOAT__ 4
#define __SIZEOF_INT128__ 16
#define __SIZEOF_INT__ 4
#define __SIZEOF_LONG_DOUBLE__ 16
#define __SIZEOF_LONG_LONG__ 8
#define __SIZEOF_LONG__ 8
#define __SIZEOF_POINTER__ 8
#define __SIZEOF_PTRDIFF_T__ 8
#define __SIZEOF_SHORT__ 2
#define __SIZEOF_SIZE_T__ 8
#define __SIZEOF_WCHAR_T__ 4
#define __SIZEOF_WINT_T__ 4
#define __SIZE_FMTX__ "lX"
#define __SIZE_FMTo__ "lo"
#define __SIZE_FMTu__ "lu"
#define __SIZE_FMTx__ "lx"
#define __SIZE_MAX__ 18446744073709551615UL
#define __SIZE_TYPE__ long unsigned int
#define __SIZE_WIDTH__ 64
#define __SSE2_MATH__ 1
#define __SSE2__ 1
#define __SSE3__ 1
#define __SSE_MATH__ 1
#define __SSE__ 1
#define __SSP__ 1
#define __SSSE3__ 1
#define __STDC_HOSTED__ 1
#define __STDC_UTF_16__ 1
#define __STDC_UTF_32__ 1
#define __STDC_VERSION__ 201112L
#define __STDC__ 1
#define __UINT16_C_SUFFIX__
#define __UINT16_FMTX__ "hX"
#define __UINT16_FMTo__ "ho"
#define __UINT16_FMTu__ "hu"
#define __UINT16_FMTx__ "hx"
#define __UINT16_MAX__ 65535
#define __UINT16_TYPE__ unsigned short
#define __UINT32_C_SUFFIX__ U
#define __UINT32_FMTX__ "X"
#define __UINT32_FMTo__ "o"
#define __UINT32_FMTu__ "u"
#define __UINT32_FMTx__ "x"
#define __UINT32_MAX__ 4294967295U
#define __UINT32_TYPE__ unsigned int
#define __UINT64_C_SUFFIX__ ULL
#define __UINT64_FMTX__ "llX"
#define __UINT64_FMTo__ "llo"
#define __UINT64_FMTu__ "llu"
#define __UINT64_FMTx__ "llx"
#define __UINT64_MAX__ 18446744073709551615ULL
#define __UINT64_TYPE__ long long unsigned int
#define __UINT8_C_SUFFIX__
#define __UINT8_FMTX__ "hhX"
#define __UINT8_FMTo__ "hho"
#define __UINT8_FMTu__ "hhu"
#define __UINT8_FMTx__ "hhx"
#define __UINT8_MAX__ 255
#define __UINT8_TYPE__ unsigned char
#define __UINTMAX_C_SUFFIX__ UL
#define __UINTMAX_FMTX__ "lX"
#define __UINTMAX_FMTo__ "lo"
#define __UINTMAX_FMTu__ "lu"
#define __UINTMAX_FMTx__ "lx"
#define __UINTMAX_MAX__ 18446744073709551615UL
#define __UINTMAX_TYPE__ long unsigned int
#define __UINTMAX_WIDTH__ 64
#define __UINTPTR_FMTX__ "lX"
#define __UINTPTR_FMTo__ "lo"
#define __UINTPTR_FMTu__ "lu"
#define __UINTPTR_FMTx__ "lx"
#define __UINTPTR_MAX__ 18446744073709551615UL
#define __UINTPTR_TYPE__ long unsigned int
#define __UINTPTR_WIDTH__ 64
#define __UINT_FAST16_FMTX__ "hX"
#define __UINT_FAST16_FMTo__ "ho"
#define __UINT_FAST16_FMTu__ "hu"
#define __UINT_FAST16_FMTx__ "hx"
#define __UINT_FAST16_MAX__ 65535
#define __UINT_FAST16_TYPE__ unsigned short
#define __UINT_FAST32_FMTX__ "X"
#define __UINT_FAST32_FMTo__ "o"
#define __UINT_FAST32_FMTu__ "u"
#define __UINT_FAST32_FMTx__ "x"
#define __UINT_FAST32_MAX__ 4294967295U
#define __UINT_FAST32_TYPE__ unsigned int
#define __UINT_FAST64_FMTX__ "lX"
#define __UINT_FAST64_FMTo__ "lo"
#define __UINT_FAST64_FMTu__ "lu"
#define __UINT_FAST64_FMTx__ "lx"
#define __UINT_FAST64_MAX__ 18446744073709551615UL
#define __UINT_FAST64_TYPE__ long unsigned int
#define __UINT_FAST8_FMTX__ "hhX"
#define __UINT_FAST8_FMTo__ "hho"
#define __UINT_FAST8_FMTu__ "hhu"
#define __UINT_FAST8_FMTx__ "hhx"
#define __UINT_FAST8_MAX__ 255
#define __UINT_FAST8_TYPE__ unsigned char
#define __UINT_LEAST16_FMTX__ "hX"
#define __UINT_LEAST16_FMTo__ "ho"
#define __UINT_LEAST16_FMTu__ "hu"
#define __UINT_LEAST16_FMTx__ "hx"
#define __UINT_LEAST16_MAX__ 65535
#define __UINT_LEAST16_TYPE__ unsigned short
#define __UINT_LEAST32_FMTX__ "X"
#define __UINT_LEAST32_FMTo__ "o"
#define __UINT_LEAST32_FMTu__ "u"
#define __UINT_LEAST32_FMTx__ "x"
#define __UINT_LEAST32_MAX__ 4294967295U
#define __UINT_LEAST32_TYPE__ unsigned int
#define __UINT_LEAST64_FMTX__ "lX"
#define __UINT_LEAST64_FMTo__ "lo"
#define __UINT_LEAST64_FMTu__ "lu"
#define __UINT_LEAST64_FMTx__ "lx"
#define __UINT_LEAST64_MAX__ 18446744073709551615UL
#define __UINT_LEAST64_TYPE__ long unsigned int
#define __UINT_LEAST8_FMTX__ "hhX"
#define __UINT_LEAST8_FMTo__ "hho"
#define __UINT_LEAST8_FMTu__ "hhu"
#define __UINT_LEAST8_FMTx__ "hhx"
#define __UINT_LEAST8_MAX__ 255
#define __UINT_LEAST8_TYPE__ unsigned char
#define __USER_LABEL_PREFIX__ _
#define __VERSION__ "4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)"
#define __WCHAR_MAX__ 2147483647
#define __WCHAR_TYPE__ int
#define __WCHAR_WIDTH__ 32
#define __WINT_TYPE__ int
#define __WINT_WIDTH__ 32
#define __amd64 1
#define __amd64__ 1
#define __apple_build_version__ 8000042
#define __block __attribute__((__blocks__(byref)))
#define __clang__ 1
#define __clang_major__ 8
#define __clang_minor__ 0
#define __clang_patchlevel__ 0
#define __clang_version__ "8.0.0 (clang-800.0.42.1)"
#define __core2 1
#define __core2__ 1
#define __llvm__ 1
#define __nonnull _Nonnull
#define __null_unspecified _Null_unspecified
#define __nullable _Nullable
#define __pic__ 2
#define __strong
#define __tune_core2__ 1
#define __unsafe_unretained
#define __weak __attribute__((objc_gc(weak)))
#define __x86_64 1
#define __x86_64__ 1
```

**Comment by gabyx** [comment]
Try to fix it

**Comment by acki-m** [comment]
@gabyx 
Take a look here:
https://github.com/rttrorg/rttr/blob/master/src/rttr/detail/base/core_prerequisites.h#L56

**Comment by gabyx** [comment]
But that only means we are on macOs, we can still use gcc there.
Committed and it seems to work on my 8.0.0 Apple Clang (at 27%)... see #100

**Comment by gabyx** [comment]
is there an include `<string>` missing in https://github.com/rttrorg/rttr/blob/29e6ff94cfd11e066cb00aed58544842534fc75b/src/rttr/detail/misc/standard_types.cpp#L34

**Comment by gabyx** [comment]
@acki-m Could you give me access such that I can kill the travis builds for my PR...?

**Comment by gabyx** [comment]
@acki-m : Success full AppleClang build here on macOs 10.11.6  with AppleClang 8.0.0 (unit tests ok!)

**Comment by acki-m** [comment]
How can I give you access to travis builds? Didn't found anything.

**Comment by acki-m** [comment]
@gabyx 
I guess we finally might have something to merge
 > Good things come to those who wait.

File: src/rttr/detail/misc/standard_types.cpp [hunk 1]
```diff
@@ -90,7 +90,8 @@ RTTR_REGISTRATION
                 .method("operator[]",   rttr::select_overload<char&(size_t)>(&std::string::operator[]))
                 .method("operator[]",   rttr::select_non_const(&std::string::operator[]))
 #endif
-#if __cplusplus >= 201703L || (RTTR_COMPILER == RTTR_COMPILER_MSVC && !defined(RTTR_NO_CXX17_NOEXCEPT_FUNC_TYPE))
+#if __cplusplus >= 201703L || (RTTR_COMPILER == RTTR_COMPILER_MSVC && !defined(RTTR_NO_CXX17_NOEXCEPT_FUNC_TYPE)) || \
+     (RTTR_COMP_VER >= 900 && RTTR_PLATFORM == RTTR_PLATFORM_APPLE)
```
Comment (gabyx) [inline src/rttr/detail/misc/standard_types.cpp]: Just out of curiosity: Why does that fix on macOs? is there no string::data method? Strange =)

**Comment by gabyx** [comment]
I merged your PR into mine: the only differences they have now is: mine builds everything on every platform -> because of different compiler warnings for benchmarks and tests. 
and that I am using a different AppleClang #if...

**Comment by gabyx** [comment]
@acki-m: I think if you add me as a collaborator in your master repo. I will not push anything, you can also restrict that I think. Maybe then I have travis rights.. .-)

**Comment by acki-m** [comment]
@gabyx 
You have got an invite :wave:
We can have a clang comiler on apple platform, which has the normal LLVM version scheme?
Hallelujah, when will this finally end.

Another approach would be, generating these kind of check by cmake.
E.g. testing, whether the warning level works, then generating a header file with the `#defines`

Anyway, we should really come to an end here and finally merge the changes to master, so that we get something. No more big changes.
I could also make MSVC2017 compile, but that would require updating the dependencies.
What do you suggest should at least work now?

**Comment by gabyx** [comment]
For sure we should finish up, mine branch compiles too (travis sometimes just refuses to apt-get packages..., but its working as you can see)
Your are correct, you can build your own LLVM clang (the normal one, prefere this one) instead of the custom apple one.
I would recommend merging my branch, since its the very same as yours (sent you a PR https://github.com/acki-m/rttr/pull/1/files)
except it uses my set_warning_leve function which lets benchmarks compile too (see travis, they work, gcc4, )
MSVC2017 doesnt work, I did not see that, I dont care so much about MSVC (this compiler sucks anyway, do they now support all standart features, extended sfinae and so on, I dont know...)
We can still in a later step, when people want to use it in MSVC17, fix the MSVC builds...

Would you be happy with this? I think the changes in both PRs are perfect =)

**Comment by acki-m** [comment]
@gabyx 
Can you disable the builds on travis now?

**Comment by gabyx** [comment]
on #100 : I am right now testing, if the os x build works, if so, I will rechange the order and it would be final for merging

**Comment by acki-m** [comment]
rejected PR, in favor of PR: #100

**Pull request closed without merging**